chain-abstraction/interfaces.go

```diff
@@ -147,13 +148,15 @@ type AssertionChain interface {
 	IsChallengeComplete(ctx context.Context, challengeParentAssertionHash AssertionHash) (bool, error)
 	Backend() ChainBackend
 	DesiredHeaderU64(ctx context.Context) (uint64, error)
+	DesiredL1HeaderU64(ctx context.Context) (uint64, error)
```

Comment:
`DesiredParentHeaderU64`

Reply:
It returns L1 block only irrespective of chain being l2 or l3, which is what we want in this case, confirmed from old staker.

Reply:
Okay. This is looking good. Sorry to be so slow on the uptake here.